src/history/historyeventoptions.ts

```diff
@@ -0,0 +1,17 @@
+export class HistoryEventOptions {
+    public details?: string;
```

Comment:
are all of these properties really optional?

Reply:
On different derivatives of HistoryEvent, yes. It's a.. bit of a property bag. I could stand to refactor this out into some interfaces.